### ISSUE TITLE
Hide bot comments in PR

### ISSUE THREAD

**Issue by jerone** [refined-github/refined-github#6213]
### Description

Comments to Pull Requests from third-party bots are very common. They are value for contributors and PR authors. As an viewer however, those comments are noise and require me to scroll to value review comments.

I would like to propose to hide* comments from bots in PRs, when
1. I'm not repo owner.
2. I'm not repo contributor.
3. I'm not repo member.
4. I'm not PR author (for first-time contributors).

*Possible ways to hide comments from bots:
1. Collapse.
2. Grey-out.
3. Opaque.
4. Hide completely, aldo I would not suggest this.

### Example URLs

`https://github.com/typescript-eslint/typescript-eslint/pull/6237`
Hide comments from typescript-eslint, nx-cloud, netlify, codecov.

**Comment by fregante** [comment]
Related:

- https://github.com/refined-github/refined-github/issues/2672#issuecomment-1208916485

**Comment by fregante** [comment]
My current solution for this:

```css
.timeline-comment:not(:hover):has(
	.author + .Label,
	.author:is(
		[href="/apps/github-actions"],
		[href="/codecov-commenter"]
	)
) {
	overflow: hidden;
	height: 39px;
}
```

<img width="848" alt="Screen Shot 3" src="https://user-images.githubusercontent.com/1402241/209616022-adc45bdb-cd3c-4773-8fde-746dc21469fa.png">

**Comment by zoey-on-github** [comment]
> My current solution for this:
> 
> ```css
> .unminimized-comment:not(
>     :hover,
>     .js-discussion > .TimelineItem:first-child *
>   ):has(
>     .author + .Label,
>     .author:is([href="/apps/github-actions"], [href="/codecov-commenter"])
>   ) {
>   overflow: hidden;
>   height: 70px;
> }
> ```
> 
> <img alt="Screen Shot 3" width="848" src="https://user-images.githubusercontent.com/1402241/209616022-adc45bdb-cd3c-4773-8fde-746dc21469fa.png">

are there any issues with this? or can this just be implemented?

**Comment by fregante** [comment]
A couple of issues:

- people might be interested in that content
- it's actually rather annoying because it opens instantly on hover, and if there are 2/3 stacked on top of each other it's nearly impossible to hover the right one

**Comment by jerone** [comment]
I would love to use the same mechanism as for off-topic comments:
![image](https://user-images.githubusercontent.com/55841/215855791-b2ca01de-c311-4f60-90f9-2ffff1670e08.png)

Combined with the current [detection of bots](https://github.com/refined-github/refined-github/blob/main/source/features/dim-bots.tsx).

**Comment by zoey-on-github** [comment]
> I would love to use the same mechanism as for off-topic comments: ![image](https://user-images.githubusercontent.com/55841/215855791-b2ca01de-c311-4f60-90f9-2ffff1670e08.png)
> 
> Combined with the current [detection of bots](https://github.com/refined-github/refined-github/blob/main/source/features/dim-bots.tsx).

i think this and having it as a toggable option(toggled on by default) for those who still  want to see the comments would be a nice way to do things

**Comment by fregante** [comment]
I agree it would be nice, it's a lot of DOM though. If the dom is already on the page (likely) it would be much more manageable. Exploration needed. 

However I'm still not convinced it's a good default unfortunately, there are bots like on TypeScript's repo that give required, extended information which should not be hidden this way

**Comment by jerone** [comment]
> However I'm still not convinced it's a good default unfortunately, there are bots like on TypeScript's repo that give required, extended information which should not be hidden this way

Agree, that's why I mentioned the conditions when to hide or not:

> I would like to propose to hide* comments from bots in PRs, when
>   1. I'm not repo owner.
>   1. I'm not repo contributor.
>   1. I'm not repo member.
>   1. I'm not PR author (for first-time contributors).

So, kinda only when you are reader.

**Comment by fregante** [comment]
Those might actually be interesting exclusions, but then this still won't stop cases where I still don't care about the bots as a PR submitter 🥲 

I'm using my CSS snippet specifically to hide comments from noisy bots on repos I contribute to, so a feature with those exclusions would not only be "not useful" but also force me update my snippet to avoid conflicts 😅 

_(To reiterate: I'm ok with my own snippet because I wrote and added it, but it's not a good default for all.)_

**Comment by fregante** [comment]
To recap:

- some bots contain fundamental info and it's not safe to just collapse/hide them for everyone, even with a "summary" style like we do in `preview-hidden-comments`

Even `dim-bots` itself is relatively annoying for people who do use bots (#5993 aims to improve the UX though) and that's not even hiding any information.

My second idea would be to de-highlight bots comments to it's easier to ignore them. Would this be useful in practice though?

<img width="612" alt="Screenshot" src="https://user-images.githubusercontent.com/1402241/218669879-74f154c8-d065-4324-b197-10a5dadbd46e.png">


```css
.unminimized-comment:not(:hover, .js-discussion > .TimelineItem:first-child *):has( .author + .Label, .author:is( [href="/apps/github-actions"], [href="/codecov-commenter"] ) ) * {
color: var(--color-border-default);
}
```

**Comment by zoey-on-github** [comment]
> To recap:
> 
> * some bots contain fundamental info and it's not safe to just collapse/hide them for everyone, even with a "summary" style like we do in `preview-hidden-comments`
> 
> Even `dim-bots` itself is relatively annoying for people who do use bots (#5993 aims to improve the UX though) and that's not even hiding any information.
> 
> My second idea would be to de-highlight bots comments to it's easier to ignore them. Would this be useful in practice though?
> 
> <img alt="Screenshot" width="612" src="https://user-images.githubusercontent.com/1402241/218669879-74f154c8-d065-4324-b197-10a5dadbd46e.png">
> 
> ```css
> .unminimized-comment:not(:hover, .js-discussion > .TimelineItem:first-child *):has( .author + .Label, .author:is( [href="/apps/github-actions"], [href="/codecov-commenter"] ) ) * {
> color: var(--color-border-default);
> }
> ```

i think we should de-highlight bot comments until you hover over them

**Comment by jd-solanki** [comment]
Can we implement this: https://github.com/bluwy/refined-github-comments

**Comment by fregante** [comment]
That userscript looks great and you can use it directly.

**Comment by fregante** [comment]
Duplicate of https://github.com/refined-github/refined-github/issues/2672

**Comment by fregante** [comment]
Ultimately this is incompatible with Refined GitHub:

- some `(bot)` comments are important
- Refined GitHub won't add a "Hide this user" option

Basically this makes the feature request the same as:

- https://github.com/refined-github/refined-github/issues/2672

So now you have at least two options:

- https://github.com/bluwy/refined-github-comments
- The CSS-only solution I [previously suggested](https://github.com/refined-github/refined-github/issues/2672#issuecomment-1208916485)

**Comment by Cocola** [comment]
For anyone still looking for a solution here — I built a lightweight Chrome extension that tackles this specific problem: [PR Zen](https://github.com/Cocola/pr-zen).

Instead of hiding bot comments (which, as @fregante rightly pointed out, can contain important info), it adds a small floating pill that classifies the PR timeline into reviews, bots, and alerts. You get a quick overview and can jump directly to what matters.

I originally built this for myself — dealing with long PRs at work where scrolling past dozens of CI and bot comments to find actual reviews was eating up my time. With the rise of AI code reviewers (Copilot, CodeRabbit, Claude…), this is only getting noisier.

It's free, open-source, and zero tracking. Feedback and contributions are very welcome — I'd love to know if this is useful to others too.
